pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -488,6 +509,9 @@ func validateFinalTasks(finalTasks []PipelineTask) *apis.FieldError {
 		if len(f.Conditions) != 0 {
 			return apis.ErrInvalidValue(fmt.Sprintf("no conditions allowed under spec.finally, final task %s has conditions specified", f.Name), "spec.finally")
 		}
+		if len(f.WhenExpressions) != 0 {
```

Comment:
@pritidesai confirming that we don't want to have when expressions in finally at this point...

Reply:
Yes, no `when` expression in finally yet 👍  `finally` does not support/allow specifying conditions and will keep them as is until we find any use cases